### PR TITLE
adding /stac catalog entry

### PR DESCRIPTION
**Pull request by hgs-trutherford**
Adding a catalog json endpoint to allow stac crawlers to interact with a dynamic catalog.

### CODE REVIEW

**Comment by hgs-trutherford** [comment]
I took an initial stab at creating the catalog schema from the example file, but I think it should be updated based on the feedback from the sprint.

File: api-spec/STAC-standalone.yaml [hunk 1]
```diff
@@ -349,6 +362,51 @@ components:
 
         extent or all relevant temporal properties.
       example: '2018-02-12T00:00:00Z/2018-03-18T12:31:12Z'
+    catalogDefinition:
+      type: object
+      properties:
```
Comment (m-mohr) [inline api-spec/STAC-standalone.yaml]: We should add the required fields:
```
required:
- name
- description 
- links
```

File: api-spec/STAC-standalone.yaml [hunk 2]
```diff
@@ -349,6 +362,51 @@ components:
...
+        name:
+          type: string
+          example: NAIP
+        description:
+          type: string
+          example: Catalog of NAIP Imagery
+        license: 
+          type: string
+          example: PDDL-1.0
+        links:
```
Comment (m-mohr) [inline api-spec/STAC-standalone.yaml]: We should add `additionalProperties: true` to allow additional properties like type or title.

File: api-spec/STAC-standalone.yaml [hunk 3]
```diff
@@ -349,6 +362,51 @@ components:
...
+          items:
+            type: string
+        keywords:
+          type: array
+          items:
+            type: string
+        homepage:
+          type: string
+        provider:
+          type: object
```
Comment (m-mohr) [inline api-spec/STAC-standalone.yaml]: `additionalProperties: true` should be added, too.

File: api-spec/STAC-standalone.yaml [hunk 4]
```diff
@@ -21,7 +21,20 @@ servers:
   - url: 'http://www.cool-sat.com'
     description: Production server
 paths:
-  /search/stac:
+  /stac:
```
Comment (m-mohr) [inline api-spec/STAC-standalone.yaml]: The geometry in line 442 could also refer to the GEOJSON geometry, i.e.
```
        geometry:
          $ref: "http://json.schemastore.org/geojson.json#/definitions/geometry"
```
But that's a nice to have and by no way blocking. May also just be solved with #131.

File: api-spec/STAC-standalone.yaml (hunk 2 above)
Comment (hgs-trutherford) [reply api-spec/STAC-standalone.yaml]: I'm a little confused on where you wan't me to add additionalProperties here. license is a string and links is an array. I am adding it at the catalogDefinition level which I think is what you wanted on your comment below as well?

Comment (m-mohr) [reply api-spec/STAC-standalone.yaml]: Was meant for the links object, seems I marked the wrong line, sorry.

**Comment by hgs-trutherford** [comment]
adding suggested changes
 @m-mohr is that what you meant with additional link properties?

**Comment by m-mohr** [comment]
It actually also makes sense in the catalog object where you put it, but one of them I was expecting in the provider object. I just added that. If there is some reason against it, just let me know. Other than that, great changes. Can be merged.

**Comment by cholmes** [comment]
Awesome, merging in.

**Pull request merged**